RELEASE_CHECKLIST.md

```diff
@@ -0,0 +1,34 @@
+# Release Checklist
+
+This is a list of the things that need to happen during a release.
+
+1. Open the associated milestone. All issues and PRs should be closed. If
+  they are not you should reassign all open issues and PRs to future
+  milestones.
+1. Go through the commit history since last release. Ensure that all PRs
+  that have landed are marked with the milestone.
```

Comment:
FWIW: https://github.com/issues?utf8=%E2%9C%93&q=repo%3Arustwasm%2Fwasm-pack+merged%3A%3E%3DYYYY-MM-DD will show all PRs merged on or after YYYY-MM-DD, which is probably a bit easier to use than CLI history for adding labels. Could be good to have the link right here for people going through the checklist.

Reply:
Sometimes stuff gets pushed right to master so that's why I look at the commit history... but we should really all avoid that :) Adding this- and thanks so much for sharing!

Reply:
Agreed -- if something gets pushed directly to master then there's no PR to add assign a milestone to anyways ;-p